nixos/lib/test-driver/test_driver/logger.py

```diff
@@ -10,6 +11,56 @@
 from xml.sax.xmlreader import AttributesImpl
 
 from colorama import Fore, Style
+from junit_xml import TestCase, TestSuite
+
+
+def _eprint(*args: object, **kwargs: Any) -> None:
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def maybe_prefix(message: str, attributes: Dict[str, str]) -> str:
+    if "machine" in attributes:
+        return f"{attributes['machine']}: {message}"
+    return message
+
+
+class BaseLogger:
+    def __init__(self) -> None:
+        self._print_serial_logs = True
+        self.logger = Logger()
+        atexit.register(self.close)
+
+    def close(self) -> None:
+        self.logger.close()
+
+    def set_logging_type(self, log_type: str) -> None:
+        if log_type == "junit-xml":
+            self.logger = JunitXMLLogger()  # type: ignore
+        else:
+            self.logger = Logger()  # type: ignore
```

Comment:
I think the general idea of hooking this feature into the existing logging infrastructure is great, and I don't see a better place to do it. But I would expect the implementation to be a bit cleaner if we built this into the existing `Logger` class, enabling Junit XML logging in addition to ordinary logging. 

What do you think?

Reply:
The existing logging class already contains features to export XML (but not a junit xml). I thought bloating up this class with another XML export would decrease readability.

Or am I getting your suggestion the wrong way?